src/Control/Middleware/RateLimitMiddleware.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+namespace SilverStripe\Control\Middleware;
+
+use SilverStripe\Control\HTTPRequest;
+use SilverStripe\Control\HTTPResponse;
+use SilverStripe\Core\Cache\RateLimiter;
+use SilverStripe\ErrorPage\ErrorPage;
+use SilverStripe\ORM\FieldType\DBDatetime;
+use SilverStripe\Security\Security;
+
+class RateLimitMiddleware implements HTTPMiddleware
+{
+
+    /**
+     * @param HTTPRequest $request
+     * @param callable $delegate
+     * @return HTTPResponse
+     */
+    public function process(HTTPRequest $request, callable $delegate)
+    {
+        $limiter = new RateLimiter($this->getKeyFromRequest($request), 10, 1);
```

Comment:
I assume the max attempts and decay constructor args aren't YAML configurable because this happens before YAML config is available? Maybe note that here?

Reply:
It's just because I wanted to do it later, tbh. I'll get it in now.

Reply:
added